sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -241,9 +245,79 @@ protected internal string GetSessionFilePath()
 
             string fileName = $"{name}{version}{async}.json";
 
-            return Path.Combine(
+            var repoRoot = GetRepoRoot();
+
+            // this needs to be updated to purely relative to repo root
+            var result = Path.Combine(
                 GetSessionFileDirectory(),
                 fileName);
+
+            return Regex.Replace(result.Replace(repoRoot, String.Empty), @"^[\\/]*", string.Empty);
```

Comment:
Might be able to use this instead - https://learn.microsoft.com/en-us/dotnet/api/system.io.path.getrelativepath?view=net-6.0

Reply:
Very possible, I'll give it a shot. The test-proxy [is a lot better](https://github.com/Azure/azure-sdk-tools/pull/4578) about handling `\` vs `/` now 👍.

Reply:
![image](https://user-images.githubusercontent.com/45376673/201211909-741e16cb-1414-4271-8b16-66c25f72bbdc.png)

Hmmm. Do we need the test framework to be cross-compatible? If so, We'll probably keep the current. I'm pretty certain this is actually why I didn't use GetRelativePath to begin with actually.

Reply:
Ah ok. Yes it would need to be able to compile against all target frameworks.